### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by XY2323819551** [facebookresearch/DPR#162]
Hi，when I used "indexer=hnsw_sq" or "indexer=hnsw"，the following error occurred，could you help me?

(/data/zhangxy/condenv/torch15DPR) zhangxy@gpu:/data/zhangxy/QA/DPR$ python dense_retriever.py \
> model_file=/data/zhangxy/QA/DPR/dpr/data/downloads/checkpoint/retriever/single/nq/bert-base-encoder.cp \
> qa_dataset=nq_test \
> indexer=hnsw_sq \
> ctx_datatsets=[dpr_wiki] \
> encoded_ctx_files=["/data/zhangxy/QA/DPR/dpr/data/downloads/data/retriever_results/nq/single/wiki_passages_*"] \
> out_file=/data/zhangxy/QA/DPR/dpr/nq-retriever-eval-top
2021-06-14 11:49:18,299 [INFO] faiss.loader: Loading faiss with AVX2 support.
2021-06-14 11:49:18,299 [INFO] faiss.loader: Could not load library with AVX2 support due to:
ModuleNotFoundError("No module named 'faiss.swigfaiss_avx2'",)
2021-06-14 11:49:18,299 [INFO] faiss.loader: Loading faiss.
2021-06-14 11:49:18,316 [INFO] faiss.loader: Successfully loaded faiss.
[2021-06-14 11:49:18,881][root][INFO] - args.local_rank -1
[2021-06-14 11:49:18,881][root][INFO] - WORLD_SIZE None
[2021-06-14 11:49:19,561][root][INFO] - Initialized host gpu as d.rank -1 on device=cuda, n_gpu=1, world size=1
[2021-06-14 11:49:19,561][root][INFO] - 16-bits training: False
[2021-06-14 11:49:19,562][root][INFO] - CFG (after gpu  configuration):
[2021-06-14 11:49:19,569][root][INFO] - indexers:
  flat:
    _target_: dpr.indexer.faiss_indexers.DenseFlatIndexer
  hnsw:
    _target_: dpr.indexer.faiss_indexers.DenseHNSWFlatIndexer
  hnsw_sq:
    _target_: dpr.indexer.faiss_indexers.DenseHNSWSQIndexer
qa_dataset: nq_test
ctx_datatsets:
- dpr_wiki
encoded_ctx_files:
- /data/zhangxy/QA/DPR/dpr/data/downloads/data/retriever_results/nq/single/wiki_passages_*
out_file: /data/zhangxy/QA/DPR/dpr/nq-retriever-eval-top
match: string
n_docs: 100
validation_workers: 16
batch_size: 128
do_lower_case: true
encoder_path: null
index_path: null
kilt_out_file: null
model_file: /data/zhangxy/QA/DPR/dpr/data/downloads/checkpoint/retriever/single/nq/bert-base-encoder.cp
validate_as_tables: false
rpc_retriever_cfg_file: null
indexer: hnsw_sq
special_tokens: null
local_rank: -1
global_loss_buf_sz: 150000
device: cuda
distributed_world_size: 1
no_cuda: false
n_gpu: 1
fp16: false
fp16_opt_level: O1
encoder:
  encoder_model_type: hf_bert
  pretrained_model_cfg: bert-base-uncased
  pretrained_file: null
  projection_dim: 0
  sequence_length: 256
  dropout: 0.1
  fix_ctx_encoder: false
  pretrained: true
datasets:
  nq_test:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.nq-test
  nq_train:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.nq-train
  nq_dev:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.nq-dev
  trivia_test:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.trivia-test
  trivia_train:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.trivia-train
  trivia_dev:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.trivia-dev
  webq_test:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.webq-test
  curatedtrec_test:
    _target_: dpr.data.retriever_data.CsvQASrc
    file: data.retriever.qas.curatedtrec-test
ctx_sources:
  dpr_wiki:
    _target_: dpr.data.retriever_data.CsvCtxSrc
    file: data.wikipedia_split.psgs_w100
    id_prefix: 'wiki:'

[2021-06-14 11:49:19,569][root][INFO] - Reading saved model from /data/zhangxy/QA/DPR/dpr/data/downloads/checkpoint/retriever/single/nq/bert-base-encoder.cp
[2021-06-14 11:49:20,066][root][INFO] - model_state_dict keys odict_keys(['model_dict', 'optimizer_dict', 'scheduler_dict', 'offset', 'epoch', 'encoder_params'])
[2021-06-14 11:49:20,071][transformers.file_utils][INFO] - PyTorch version 1.5.1 available.
[2021-06-14 11:49:21,667][transformers.configuration_utils][INFO] - loading configuration file https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-uncased-config.json from cache at /home/zhangxy/.cache/torch/transformers/4dad0251492946e18ac39290fcfe91b89d370fee250efe9521476438fe8ca185.7156163d5fdc189c3016baca0775ffce230789d7fa2a42ef516483e4ca884517
[2021-06-14 11:49:21,668][transformers.configuration_utils][INFO] - Model config BertConfig {
  "architectures": [
    "BertForMaskedLM"
  ],
  "attention_probs_dropout_prob": 0.1,
  "gradient_checkpointing": false,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 768,
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "layer_norm_eps": 1e-12,
  "max_position_embeddings": 512,
  "model_type": "bert",
  "num_attention_heads": 12,
  "num_hidden_layers": 12,
  "pad_token_id": 0,
  "type_vocab_size": 2,
  "vocab_size": 30522
}

[2021-06-14 11:49:22,803][transformers.modeling_utils][INFO] - loading weights file https://cdn.huggingface.co/bert-base-uncased-pytorch_model.bin from cache at /home/zhangxy/.cache/torch/transformers/f2ee78bdd635b758cc0a12352586868bef80e47401abe4c4fcc3832421e7338b.36ca03ab34a1a5d5fa7bc3d03d55c4fa650fed07220e2eeebc06ce58d0e9a157
[2021-06-14 11:49:26,519][transformers.modeling_utils][INFO] - All model checkpoint weights were used when initializing HFBertEncoder.

[2021-06-14 11:49:26,520][transformers.modeling_utils][INFO] - All the weights of HFBertEncoder were initialized from the model checkpoint at bert-base-uncased.
If your task is similar to the task the model of the ckeckpoint was trained on, you can already use HFBertEncoder for predictions without further training.
[2021-06-14 11:49:27,697][transformers.configuration_utils][INFO] - loading configuration file https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-uncased-config.json from cache at /home/zhangxy/.cache/torch/transformers/4dad0251492946e18ac39290fcfe91b89d370fee250efe9521476438fe8ca185.7156163d5fdc189c3016baca0775ffce230789d7fa2a42ef516483e4ca884517
[2021-06-14 11:49:27,698][transformers.configuration_utils][INFO] - Model config BertConfig {
  "architectures": [
    "BertForMaskedLM"
  ],
  "attention_probs_dropout_prob": 0.1,
  "gradient_checkpointing": false,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 768,
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "layer_norm_eps": 1e-12,
  "max_position_embeddings": 512,
  "model_type": "bert",
  "num_attention_heads": 12,
  "num_hidden_layers": 12,
  "pad_token_id": 0,
  "type_vocab_size": 2,
  "vocab_size": 30522
}

[2021-06-14 11:49:29,365][transformers.modeling_utils][INFO] - loading weights file https://cdn.huggingface.co/bert-base-uncased-pytorch_model.bin from cache at /home/zhangxy/.cache/torch/transformers/f2ee78bdd635b758cc0a12352586868bef80e47401abe4c4fcc3832421e7338b.36ca03ab34a1a5d5fa7bc3d03d55c4fa650fed07220e2eeebc06ce58d0e9a157
[2021-06-14 11:49:33,078][transformers.modeling_utils][INFO] - All model checkpoint weights were used when initializing HFBertEncoder.

[2021-06-14 11:49:33,079][transformers.modeling_utils][INFO] - All the weights of HFBertEncoder were initialized from the model checkpoint at bert-base-uncased.
If your task is similar to the task the model of the ckeckpoint was trained on, you can already use HFBertEncoder for predictions without further training.
[2021-06-14 11:49:34,922][transformers.tokenization_utils_base][INFO] - loading file https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-uncased-vocab.txt from cache at /home/zhangxy/.cache/torch/transformers/26bc1ad6c0ac742e9b52263248f6d0f00068293b33709fae12320c0e35ccfbbb.542ce4285a40d23a559526243235df47c5f75c197f04f37d1a0c124c32c9a084
[2021-06-14 11:49:34,980][root][INFO] - Selecting standard question encoder
[2021-06-14 11:49:36,465][root][INFO] - Loading saved model state ...
[2021-06-14 11:49:36,466][root][INFO] - Encoder state prefix question_model.
[2021-06-14 11:49:36,645][root][INFO] - Encoder vector_size=768
[2021-06-14 11:49:36,645][root][INFO] - qa_dataset: nq_test
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - Requested resource from https://dl.fbaipublicfiles.com/dpr/data/retriever/nq-test.qa.csv
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - Download root_dir /data/zhangxy/QA/DPR
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - File to be downloaded as /data/zhangxy/QA/DPR/downloads/data/retriever/qas/nq-test.csv
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - File already exist /data/zhangxy/QA/DPR/downloads/data/retriever/qas/nq-test.csv
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - Loading from https://dl.fbaipublicfiles.com/dpr/nq_license/LICENSE
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - File already exist /data/zhangxy/QA/DPR/downloads/data/retriever/qas/LICENSE
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - Loading from https://dl.fbaipublicfiles.com/dpr/nq_license/README
[2021-06-14 11:49:36,675][dpr.data.download_data][INFO] - File already exist /data/zhangxy/QA/DPR/downloads/data/retriever/qas/README
[2021-06-14 11:49:36,730][root][INFO] - Index class <class 'dpr.indexer.faiss_indexers.DenseHNSWSQIndexer'>
[2021-06-14 11:49:36,731][root][INFO] - Using special token None
[2021-06-14 11:50:00,655][root][INFO] - Encoded queries 3200
[2021-06-14 11:50:03,712][root][INFO] - Total encoded queries tensor torch.Size([3610, 768])
[2021-06-14 11:50:03,746][root][INFO] - id_prefixes per dataset: ['wiki:']
[2021-06-14 11:50:03,747][root][INFO] - ctx_files_patterns: ['/data/zhangxy/QA/DPR/dpr/data/downloads/data/retriever_results/nq/single/wiki_passages_*']
[2021-06-14 11:50:03,748][root][INFO] - Embeddings files id prefixes: []
[2021-06-14 11:50:03,748][root][INFO] - Reading all passages data from files: []
[2021-06-14 11:50:03,748][root][INFO] - HNSWF DotProduct -> L2 space phi=0
[2021-06-14 11:50:03,748][root][INFO] - Total data indexed 0
[2021-06-14 11:50:03,748][root][INFO] - Data indexing completed.
[2021-06-14 11:50:03,753][root][INFO] - query_hnsw_vectors (3610, 769)
Segmentation fault (core dumped)

**Comment by vlad-karpukhin** [comment]
Hi, 
It seems to be the same issue like in your another report you just created - embeddings files pattern seems to be incorrect and points to no real embedding files.
